pkg/template/processor.go

```
@@ -0,0 +1,193 @@
+	}
+
+	// gets the meta accessor to the existing resource
+	metaExisting, err := meta.Accessor(existing)
+	if err != nil {
+		return false, errors.Wrapf(err, "cannot get metadata from %+v", existing)
+	}
+
+	// as it already exists, check using the UpdateStrategy if it should be updated
+	if !forceUpdate {
+		existingAnnotations := metaExisting.GetAnnotations()
+		if existingAnnotations != nil {
+			if newConfiguration == existingAnnotations[LastAppliedConfigurationAnnotationKey] {
+				return false, nil
+			}
+		}
+	}
+
+	// retrieve the current 'resourceVersion' to set it in the resource passed to the `client.Update()`
+	// otherwise we would get an error with the following message:
+	// "nstemplatetiers.toolchain.dev.openshift.com \"basic\" is invalid: metadata.resourceVersion: Invalid value: 0x0: must be specified for an update"
+	originalGeneration := metaExisting.GetGeneration()
+	metaNew.SetResourceVersion(metaExisting.GetResourceVersion())
+	if err := p.cl.Update(context.TODO(), newResource); err != nil {
+		return false, errors.Wrapf(err, "unable to update the resource '%v'", newResource)
+	}
+
+	// gets the meta accessor to the resource that was updated
+	metaNewAfterUpdate, err := meta.Accessor(newResource)
+	if err != nil {
+		return false, errors.Wrapf(err, "cannot get metadata from %+v", newResource)
+	}
+
+	// check if it was changed or not
+	return originalGeneration == metaNewAfterUpdate.GetGeneration(), nil
+}
+
+func getNewConfiguration(newResource runtime.Object) string {
+	newYaml := marshalObjectContent(newResource)
+
+	if len(newYaml) == 0 {
+		return fmt.Sprintf("%v", newResource)
+	}
+	return string(newYaml)
+}
+
+func marshalObjectContent(newResource runtime.Object) []byte {
+	var newJson []byte
+	var err error
+	newRes, ok := newResource.(runtime.Unstructured)
+	if ok {
+		newJson, err = json.Marshal(newRes.UnstructuredContent())
+		if err != nil {
+			log.Error(err, "unable to marshal the object", "object", newRes.UnstructuredContent())
```

aren't you supposed to return an error instead of just logging it?

---

yes I'm :-) see 8720488